jobs/payment-jobs/tests/jobs/test_ejv_partner_partial_refund_distribution_task.py

```diff
@@ -0,0 +1,90 @@
+from unittest.mock import patch
+
+import pytest
+from flask import current_app
+from freezegun import freeze_time
+from pay_api.models import CorpType as CorpTypeModel
+from pay_api.models import DistributionCode, EjvFile, EjvHeader, EjvLink, FeeSchedule, db
+from pay_api.utils.enums import DisbursementStatus
+
+from tasks.ejv_partner_distribution_task import EjvPartnerDistributionTask
+
+from .factory import (
+    factory_create_direct_pay_account, factory_distribution, factory_distribution_link, factory_invoice,
+    factory_invoice_reference, factory_payment, factory_payment_line_item, factory_receipt)
+
+
+@pytest.mark.parametrize('client_code, batch_type', [('112', 'GA'), ('113', 'GI'), ('ABC', 'GI')])
+def test_partial_refund_disbursement_for_partners(session, monkeypatch, client_code, batch_type):
+    """Test partial refund disbursement for partners."""
+    monkeypatch.setattr('pysftp.Connection.put', lambda *args, **kwargs: None)
+    corp_type: CorpTypeModel = CorpTypeModel.find_by_code('VS')
+
+    pay_account = factory_create_direct_pay_account()
+
+    disbursement_distribution: DistributionCode = factory_distribution(name='VS Disbursement', client=client_code)
+    service_fee_distribution: DistributionCode = factory_distribution(name='VS Service Fee', client='112')
+    fee_distribution: DistributionCode = factory_distribution(
+        name='VS Fee distribution', client='112', service_fee_dist_id=service_fee_distribution.distribution_code_id,
+        disbursement_dist_id=disbursement_distribution.distribution_code_id
+    )
+    fee_schedule: FeeSchedule = FeeSchedule.find_by_filing_type_and_corp_type(corp_type.code, 'WILLNOTICE')
+    factory_distribution_link(fee_distribution.distribution_code_id, fee_schedule.fee_schedule_id)
+
+    invoice = factory_invoice(payment_account=pay_account,
+                              corp_type_code=corp_type.code, total=21.5, status_code='PAID')
+    factory_payment_line_item(invoice_id=invoice.id, fee_schedule_id=fee_schedule.fee_schedule_id,
+                              filing_fees=10, total=10, service_fees=1.5,
+                              fee_dist_id=fee_distribution.distribution_code_id)
+
+    # Partially refunded line item
+    pli_refunded = factory_payment_line_item(invoice_id=invoice.id, fee_schedule_id=fee_schedule.fee_schedule_id,
+                                             filing_fees=10, total=10, service_fees=1.5,
+                                             fee_dist_id=fee_distribution.distribution_code_id)
+
+    inv_ref = factory_invoice_reference(invoice_id=invoice.id)
+    factory_payment(invoice_number=inv_ref.invoice_number, payment_status_code='COMPLETED')
+    factory_receipt(invoice_id=invoice.id, receipt_date=datetime.today()).save()
+
+    # Mock the factory_refund_partial_line to simulate partial refund creation, cause 'RefundPartialLine' has no 'save'
+    with patch('pay_api.models.RefundPartialLine') as mock_refund_partial:
```

Comment:
not a good idea, why not just create the object like the pattern above?

Reply:
eg. `factory_partial_refund()`

Reply:
`RefundsPartial` is what you want to create not `RefundPartialLine` (dataclass after the model)

If you need `.save` it's in the base model that you could inherit:

BaseModel

```
    def save(self):
        """Save and commit."""
        db.session.add(self)
        db.session.flush()
        self.create_activity(self)
        db.session.commit()
        return self
```

Reply:
Thanks for point it out, I added:
```
def factory_refund_partial(
        payment_line_item_id: int,
        refund_amount: float,
        refund_type: str,
        created_by='test',
        created_on: datetime = datetime.now()
):
    """Return Factory."""
    return RefundsPartial(
        payment_line_item_id=payment_line_item_id,
        refund_amount=refund_amount,
        refund_type=refund_type,
        created_by=created_by,
        created_on=created_on
    ).save() 
```